client/js/test/cache_map_test.js

```diff
@@ -0,0 +1,402 @@
+/*
+ * Copyright 2014 Treode, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+	To run: cd to /client/js/ and run 'mocha'
+*/
+var cache_map = require('../src/cache_map.js');
+var test = require('unit.js');
+
+
+//cache_map put tests
+describe('cache_map.put()',function(){
+
+	//most basic test, should accept 3 puts and all elements should be in the cache map
+	//does not validate values, that is done in 'get' tests
+	it('should place 3 objects with different keys in the cache ', function(done){
```

Comment:
It looks like if you leave `done` out of the args, then mocha will run a synchronous test instead of an asynchronous one.

http://mochajs.org/#synchronous-code